src/data-models/mdm-data-class.resource.ts

```diff
@@ -499,4 +499,19 @@ export class MdmDataClassResource extends MdmResource {
     const url = `${this.apiEndpoint}/dataModels/${dataModelId}/dataClasses/${dataClassId}/extends/${otherDataModelId}/${otherDataClassId}`;
     return this.simpleDelete(url, {}, options);
   }
+
+  /*
+    * `HTTP POST` - Copy a data class from a source data model to a target data model.
+    *
+    * @param CopyDataClassPyaload The payload of the request containing the source and target data model and data class IDs.
+    * @returns The result of the `POST` request.
+    *
+    * `200 OK` - will return a {@link DataClassDetailResponse} containing a {@link DataClassDetail} object.
+  */
+
+  copy(dataClassId: Uuid, copyDataClassPayload: CopyDataClassPayload){
```

Comment:
I think it's interesting that the parameters are grouped together. But I also think there are inconsistencies:

1. Other methods have a final `options?: RequestSettings` parameter, but this does not.
2. All the parameters in the `CopyDataClassPayload` are being passed in the `POST` request payload, not just the necessary parameters.
3. `dataClassId` is the source data class ID, but other properties are clearly labelled `source...`

I think to remain consistent with the rest of `mdm-resources`:

- [ ] Move the `targetDataModelId`, `sourceDataModelId` properties as parameters to this method instead.
- [ ] The `CopyDataClassPayload` should only contain `copyLabel` and `copyPermissions`
- [ ] Add the missing optional `options:? RequestSettings` parameter